### ISSUE TITLE
"no connection available" when using DiscoverNodesOnStart

### ISSUE THREAD

**Issue by lcosmin** [elastic/go-elasticsearch#256]
I'm using the following [code snippet](https://play.golang.com/p/LH7ruln6-Ne) to read documents from Elasticsearch.

If I enable node discovery on start, I end up getting a `no connection available` error on my test cluster, apparently due to the fact that all cluster nodes are have both `data` and `master` roles.

```
2021/04/21 14:09:47 connecting to ES
2021/04/21 14:09:47 successfully connected
Discovered node [xxxxxxx]; http://10.x.y.z1:9200; roles=[data master]; [SKIP]
Discovered node [yyyyyyy]; http://10.x.y.z2:9200; roles=[data master]; [SKIP]
Discovered node [zzzzzzz]; http://10.x.y.z3:9200; roles=[data master]; [SKIP]
2021/04/21 14:09:48 getting document
2021/04/21 14:09:48 error performing request: cannot get connection: no connection available
2021/04/21 14:09:49 getting document
2021/04/21 14:09:49 error performing request: cannot get connection: no connection available
2021/04/21 14:09:50 getting document
2021/04/21 14:09:50 error performing request: cannot get connection: no connection available
^Csignal: interrupt
```

Commenting the `DiscoverNodesOnStart: true,` part allows the code to work properly.
Is this an expected behavior ?

Using `github.com/elastic/go-elasticsearch/v8 v8.0.0-20210421122141-109055da9e51` with Elasticsearch 7.10.2

**Comment by Anaethelion** [comment]
This definitely is not intended, the expected behavior is to skip master only nodes.
The `v8` client is intended to be used with the next major release, could you try with the `7.12` release instead ?

**Comment by lcosmin** [comment]
I get the same behavior with 
```
require github.com/elastic/go-elasticsearch/v7 v7.12.0
```

**Comment by lcosmin** [comment]
The problem seems to be here: [https://github.com/elastic/go-elasticsearch/blob/master/estransport/discovery.go#L79](https://github.com/elastic/go-elasticsearch/blob/master/estransport/discovery.go#L79)

My node is only "data" + "master", so it gets skipped because it's not "ingest" too. Think maybe the line there (and above) should be `if !isDataNode && !isIngestNode {`

**Comment by Anaethelion** [comment]
The fix has been released in the `7.13` version of the client, closing this issue.
